files/tk_metrics

```diff
@@ -171,7 +185,8 @@ HOSTS.each do |host|
     dataset['servers'][hostkey][METRICS_TYPE]['api-query-start'] = timestamp.utc.iso8601
     dataset['servers'][hostkey][METRICS_TYPE]['api-query-duration'] = Time.now - timestamp
 
-    json_dataset = JSON.pretty_generate(dataset)
+    filtered_dataset = filter_metrics(dataset, EXCLUDES)
+    json_dataset = JSON.pretty_generate(filtered_dataset)
```

Comment:
I did some benchmarking and it turns out that dropping the extra whitespace created by `pretty_generate` is going to save us significant amounts of space.

For a monolithic master with 8 compilers and one deployed Code Manager environment, the daily archive size was 16 MB. Removing the `file-sync-storage-service` and `pe-` datasets drops that to 13 MB.

Removing the whitespace created by `pretty_generate` and keeping the data drops the archive size to 7.8 MB.

Removing both the whitespace and the data drops it to 5.6 MB. Disk space required for uncompressed files was reduced from 200 MB to 50 MB once everything was removed.

I would expect the savings from removing `file-sync-storage-service` to increase rapidly as the number of environments grows, but killing off the whitespace is still a significant reduction in size.

Reply:
My only concern here is the ability to grep metrics without using a post-processing tool. The data is the same, but the README instructions will need to change, so I guess this could be considered a breaking change? 

I can update the instructions to use something like `jq` and the minified JSON though.  Would it be better to add instructions with `jq` or do something with ruby/another built-in library?

Reply:
Updated in the recent commits

Reply:
I feel like the volume of data is generally large enough that post processing tools are needed to summarize it. `grep` is very convenient, but you can't run `grep` on a data archive that never got uploaded because it was too large for a user to deal with effectively.

If the space savings was smaller, say 5%, then it cold be worth keeping the pretty printing. With a savings of nearly 50%, the opportunity to cut the size of the data in half seems like something we need to take given the difficulty folks have reported with wrangling large support scripts and metrics archives.

Reply:
Absolutely agreed. I updated the documentation to provide a way to enable the `grep` workflow and made the change to remove the pretty printing in the latest commits.